Common++/header/IpAddresses.h

```diff
@@ -0,0 +1,551 @@
+		 * @param[in] arr A pointer to where array will be allocated
+		 * @param[out] length Returns the length in bytes of the array that was allocated
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is now unnecessary and added for backward compatibility. It will be deleted in the future")]]
+		#endif
+		void copyTo(uint8_t** arr, size_t& length) const;
+
+		/**
+		 * Gets a pointer to an already allocated byte array and copies the address value to it.
+		 * Notice this method is deprecated and will be deleted in the future
+		 * This method assumes array allocated size is at least 16 (the size of an IPv6 address)
+		 * @param[in] arr A pointer to the array which address will be copied to
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is now unnecessary and added for backward compatibility. It will be deleted in the future")]]
+		#endif
+		void copyTo(uint8_t* arr) const;
+
+		/**
+		 * A static value representing a zero value of IPv6 address, meaning address of value "0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0"
+		 * Notice this value can be omitted in the user code because the default constructor creates an instance with an unspecified/zero address.
+		 * In order to check whether the address is zero the method isUnspecified can be used
+		 */
+		static const IPv6Address Zero;
+
+	private:
+		uint8_t m_Bytes[16];
+	}; // class IPv6Address
+
+
+	// Creation of IPv6Address
+
+	/**
+	 * A function that creates an instance of the class IPv6Address out of 16-byte array.
+	 * @param[in] bytes The address as 16-byte array in network byte order
+	 * @return An instance of class IPv6Address
+	 */
+	inline IPv6Address makeIPv6Address(const uint8_t bytes[16]) { return IPv6Address(bytes); }
+
+	/**
+	 * A function that creates an instance of the class out of string (char*) value
+	 * If the string doesn't represent a valid IPv6 address an errorCode will be set to non-zero value
+	 * @param[in] addressAsString The string (char*) representation of the address
+	 * @param[out] errorCode Contains 0 if a string represents a valid address, otherwise a non-zero value
+	 * @return An instance of class IPv6Address. If an error occured the return address will be an unspecified/zero
+	 */
+	IPv6Address makeIPv6Address(const char* addrAsString, int& errorCode);
+
+	/**
+	 * A function that creates an instance of the class out of std::string value
+	 * If the string doesn't represent a valid IPv6 address an errorCode will be set to non-zero value
+	 * @param[in] addressAsString The std::string representation of the address
+	 * @param[out] errorCode Contains 0 if a string represents a valid address, otherwise a non-zero value
+	 * @return An instance of class IPv6Address. If an error occured the return address will be an unspecified/zero
+	 */
+	inline IPv6Address makeIPv6Address(const std::string& addrAsString, int& errorCode) {	return makeIPv6Address(addrAsString.c_str(), errorCode); }
```

Comment:
see my comment regarding these methods for `IPv4Address`

Reply:
Two functions have been removed. One function has been made a static method.

Reply:
Let's have the same solution we agreed for `IPv4Address`

Reply:
Done